src/actions/search-results.js

```diff
@@ -108,54 +109,106 @@ export function clearFeed () {
   return { type: CLEAR_FEED };
 }
 
+function isTile (item) {
+  return !item.packageOffer;
+}
+
 /**
 * Buffer and show tiles in a mixed fashion
 * to be used by saveSearchResult
 */
 
 export function mixDataInput () {
   // this fn is used to setup a result store, so that we are instance specific.
+  let tiles = [];
+  let packages = [];
+
   let mixture = [];
-  let bufferedResponse = [];
+
   let steps = 1;
-  let highwatermark = 10;
+  let highwatermark = 20;
+
+  let timeout = null;
+
+  function shake () {
+    const total = tiles.length + packages.length;
+    // randomise tiles
+    shuffle(tiles);
+    // sort by rank packages
+
+    // weave tiles and packages to mixture
+    for (let i = 1; i < total + 1; i++) {
+      if (packages.length > 0) {
+        if (i % 6) {
```

Comment:
> This will mean that if there are relatively few packages compared with the number of tiles then the tiles will be sparsely spread until the end when they'll be a glut. Of course, that might be intended behaviour.

This has now been handled.

> I'd be tempted to replace the magic number 6 with Math.floor(packages.length / tiles.length) so that tiles are roughly equally spaced amongst packages, irrespective of the ratio between them.

Sounds like a plan.